common/random.cpp

```diff
@@ -48,21 +48,20 @@ RandomSource::RandomSource(const String &name) {
 }
 
 void RandomSource::setSeed(uint32 seed) {
+	if(seed == 0)
+		seed++;
 	_randSeed = seed;
 }
 
 uint RandomSource::getRandomNumber(uint max) {
-	_randSeed = 0xDEADBF03 * (_randSeed + 1);
-	_randSeed = (_randSeed >> 13) | (_randSeed << 19);
-
+	scrambleSeed();
 	if (max == UINT_MAX)
 		return _randSeed;
```

Comment:
should this also be multiplied by 0xDEADBF03 for this return?

Reply:
Fixed in a0bc664.